src/safir/github/webhooks.py

```diff
@@ -0,0 +1,343 @@
+"""Pydantic models for GitHub webhook payloads."""
+
+from __future__ import annotations
+
+from enum import Enum
+
+from pydantic import BaseModel, Field
+
+from .models import (
+    GitHubCheckRunModel,
+    GitHubCheckSuiteModel,
+    GitHubPullRequestModel,
+    GitHubRepositoryModel,
+)
+
+__all__ = [
+    "GitHubAppInstallationModel",
+    "GitHubPushEventModel",
+    "GitHubAppInstallationEventRepoModel",
+    "GitHubAppInstallationEventAction",
+    "GitHubAppInstallationEventModel",
+    "GitHubAppInstallationRepositoriesEventAction",
+    "GitHubAppInstallationRepositoriesEventModel",
+    "GitHubPullRequestEventAction",
+    "GitHubPullRequestEventModel",
+    "GitHubCheckSuiteEventAction",
+    "GitHubCheckSuiteEventModel",
+    "GitHubCheckRunEventAction",
+    "GitHubCheckRunEventModel",
+]
+
+
+class GitHubAppInstallationModel(BaseModel):
+    """A Pydantic model for the ``installation`` field found in webhook
+    payloads for GitHub Apps.
+    """
+
+    id: str = Field(description="The installation ID.")
```

Comment:
For these, I probably would add `title` since they're going to go into the OpenAPI documentation of the app that defines this webhook, but it's likely I'm being too pedantic.

Reply:
In practice I don't have these appear in the OpenAPI docs for the webhook endpoint because I defer Pydantic deserialization to the webhook handler after the Gidgethub's verification and routing steps, rather than having FastAPI deserialize the data into any one model.

Reply:
Oh! Yes, that was the piece I was missing. I'm good with omitting `title` in that case, although I'm guessing that when I add that plugin to the Gafaelfawr documentation build for better model documentation, I'm going to end up having to write a bit of duplicate documentation.